apps/server/src/modules/tool/controller/tool.controller.spec.ts

```diff
@@ -121,40 +157,153 @@ describe('ToolController', () => {
 
 			const currentUser: ICurrentUser = { userId: 'userId' } as ICurrentUser;
 
+			const customParameterResponse: CustomParameterResponse = new CustomParameterResponse({
+				name: 'mockName',
+				default: 'mockDefault',
+				location: CustomParameterLocationParams.PATH,
+				scope: CustomParameterScopeParams.SCHOOL,
+				type: CustomParameterTypeParams.STRING,
+				regex: 'mockRegex',
+			});
+			const basicToolConfigResponse: BasicToolConfigResponse = new BasicToolConfigResponse({
+				type: ToolConfigType.BASIC,
+				baseUrl: 'mockUrl',
+			});
+			const lti11ToolConfigResponse: Lti11ToolConfigResponse = new Lti11ToolConfigResponse({
+				key: 'mockKey',
+				lti_message_type: LtiMessageType.BASIC_LTI_LAUNCH_REQUEST,
+				privacy_permission: LtiPrivacyPermission.NAME,
+				type: ToolConfigType.LTI11,
+				baseUrl: 'mockUrl',
+			});
+			const oauth2ToolConfigResponse: Oauth2ToolConfigResponse = new Oauth2ToolConfigResponse({
+				clientId: 'mockId',
+				skipConsent: false,
+				type: ToolConfigType.OAUTH2,
+				baseUrl: 'mockUrl',
+			});
+			const externalToolResponse: ExternalToolResponse = new ExternalToolResponse({
+				id: '1',
+				name: 'mockName',
+				url: 'mockUrl',
+				logoUrl: 'mockLogoUrl',
+				parameters: [customParameterResponse],
+				isHidden: true,
+				openNewTab: true,
+				version: 1,
+				config: basicToolConfigResponse,
+			});
+
+			const basicToolConfigDO: BasicToolConfigDO = new BasicToolConfigDO({
+				type: ToolConfigType.BASIC,
+				baseUrl: 'mockUrl',
+			});
+			const lti11ToolConfigDO: Lti11ToolConfigDO = new Lti11ToolConfigDO({
+				secret: 'mockSecret',
+				key: 'mockKey',
+				lti_message_type: LtiMessageType.BASIC_LTI_LAUNCH_REQUEST,
+				privacy_permission: LtiPrivacyPermission.NAME,
+				type: ToolConfigType.LTI11,
+				baseUrl: 'mockUrl',
+			});
+			const oauth2ToolConfigDO: Oauth2ToolConfigDO = new Oauth2ToolConfigDO({
+				clientId: 'mockId',
+				skipConsent: false,
+				type: ToolConfigType.OAUTH2,
+				baseUrl: 'mockUrl',
+			});
+			const customParameterDO: CustomParameterDO = new CustomParameterDO({
+				name: 'mockName',
+				default: 'mockDefault',
+				location: CustomParameterLocation.PATH,
+				scope: CustomParameterScope.SCHOOL,
+				type: CustomParameterType.STRING,
+				regex: 'mockRegex',
+			});
+			const externalToolDO: ExternalToolDO = new ExternalToolDO({
+				id: '1',
+				name: 'mockName',
+				url: 'mockUrl',
+				logoUrl: 'mockLogoUrl',
+				parameters: [customParameterDO],
+				isHidden: true,
+				openNewTab: true,
+				version: 1,
+				config: basicToolConfigDO,
+			});
+
 			return {
 				body,
 				bodyConfigCreateBasicParams,
 				bodyConfigCreateOauthParams,
 				bodyConfigCreateLti11Params,
 				currentUser,
+				externalToolResponse,
```

Comment:
do all tests need all these things? maybe the function gets more managable if it is split into different setups, like setupWithOauth, setupWithLti11 or something similar (whatever allows you to still only use one setup per test).

This could be extended by adding describe blocks ("when creating an oauth tool...")

Reply:
added setups and splitted the tests, I hope it's fine now :)